src/bentoml/_internal/configuration/v2/__init__.py

```diff
@@ -0,0 +1,185 @@
+            s.Optional("max"): s.Or(s.And(float, ensure_larger_than_zero), None),
+            s.Optional("factor"): s.Or(s.And(float, ensure_larger_than(1.0)), None),
+        },
+    },
+    s.Optional("logging"): {
+        "access": {
+            "enabled": bool,
+            "request_content_length": s.Or(bool, None),
+            "request_content_type": s.Or(bool, None),
+            "response_content_length": s.Or(bool, None),
+            "response_content_type": s.Or(bool, None),
+            "format": {
+                "trace_id": str,
+                "span_id": str,
+            },
+        },
+    },
+    s.Optional("http"): {
+        "host": s.And(str, is_valid_ip_address),
+        "port": s.And(int, ensure_larger_than_zero),
+        "cors": {
+            "enabled": bool,
+            "access_control_allow_origins": s.Or([str], str, None),
+            "access_control_allow_origin_regex": s.Or(
+                s.And(str, s.Use(re.compile)), None
+            ),
+            "access_control_allow_credentials": s.Or(bool, None),
+            "access_control_allow_headers": s.Or([str], str, None),
+            "access_control_allow_methods": s.Or([str], str, None),
+            "access_control_max_age": s.Or(int, None),
+            "access_control_expose_headers": s.Or([str], str, None),
+        },
+        "response": {"trace_id": bool},
+    },
+    s.Optional("grpc"): {
+        "host": s.And(str, is_valid_ip_address),
+        "port": s.And(int, ensure_larger_than_zero),
+        "metrics": {
+            "port": s.And(int, ensure_larger_than_zero),
+            "host": s.And(str, is_valid_ip_address),
+        },
+        "reflection": {"enabled": bool},
+        "channelz": {"enabled": bool},
+        "max_concurrent_streams": s.Or(int, None),
+        "max_message_length": s.Or(int, None),
+        "maximum_concurrent_rpcs": s.Or(int, None),
+    },
+    s.Optional("ssl"): {
+        "enabled": bool,
+        s.Optional("certfile"): s.Or(str, None),
+        s.Optional("keyfile"): s.Or(str, None),
+        s.Optional("keyfile_password"): s.Or(str, None),
+        s.Optional("version"): s.Or(s.And(int, ensure_larger_than_zero), None),
+        s.Optional("cert_reqs"): s.Or(int, None),
+        s.Optional("ca_certs"): s.Or(str, None),
+        s.Optional("ciphers"): s.Or(str, None),
+    },
+    s.Optional("runner_probe"): {
+        "enabled": bool,
+        "timeout": int,
+        "period": int,
+    },
+    s.Optional("monitoring"): {
+        "enabled": bool,
+        s.Optional("type"): s.Or(str, None),
+        s.Optional("options"): s.Or(dict, None),
+    },
+    s.Optional("tracing"): TRACING_CFG,
+}
+
+SCHEMA = s.Schema(
+    {
+        s.Optional("version", default=2): s.And(int, lambda v: v == 2),
+        "services": {
+            **_SERVICE_CONFIG,
+            s.Optional(str): _SERVICE_CONFIG,
+        },
+    }
+)
+
+
+def migration(*, default_config: dict[str, t.Any], override_config: dict[str, t.Any]):
```

Comment:
Should this function be responsible for migrating v1 to v2? as the name suggests.

Reply:
The `migrate` function is used to migrate some deprecated fields in the same version. And I don't think we'll actually migrate configs from v1 to v2, the version of configs depends on the version of bentoml.